crates/ast/src/parser.rs

```diff
@@ -341,6 +341,10 @@ fn sol_type<'tokens, 'src: 'tokens>() -> impl Parser<'tokens, 'src, Spanned<DynS
     recursive(|sol_raw_type| {
         let sol_raw_primitive_type = ident().map(|(typ, _)| typ.to_string());
 
+        let sol_raw_event_primitive_type = ident()
```

Comment:
This would also allow indexed args inside tuples and inside function args. Maybe we should give the sol_type_parser a flag to enable "indexed" args.

Reply:
You're right, let me see how I can fix this.